jdaviz/app.py

```diff
@@ -1808,6 +1842,12 @@ def compose_viewer_area(viewer_area_items):
 
         for name in config.get('tray', []):
             tray = tray_registry.members.get(name)
+            tray_registry_name = tray.get('cls')._registry_name
+
+            if not self._plugin_is_compatible(tray_registry_name):
+                # skip tray item constructor if not compatible
+                continue
```

Comment:
I guess since the version requirement is a dev-version of astropy, skipping might make sense (otherwise I'd suggest using the existing disabled infrastructure so users aren't confused and we can provide instructions on what they need to upgrade, etc).  Should this requirement be moved to the plugin itself though?  Or do you see this as a temporary solution just until astropy releases?

Reply:
@kecnry Cool, I didn't know there was existing infrastructure. Is it as simple as toggling this bool? 

https://github.com/spacetelescope/jdaviz/blob/9ca998e32c3db3a0e95b4bd9b963b56937f55895/jdaviz/core/template_mixin.py#L172-L173

Reply:
And sorry, to answer your questions:

> Should this requirement be moved to the plugin itself though?

We could do that too.

> Or do you see this as a temporary solution just until astropy releases?

I saw this as either/both (1) a temporary solution until release, (2) planting a seed in case someone else needs it later. Not super attached to it, take it or leave it 👍🏻

Reply:
It should be as simple as setting the `disabled_msg` traitlet (in the plugin's init, for example) to anything other than an empty string.  The plugin will still show in the tray, but when opening will only display that message.  I think this would be useful to say "spectral extraction requires astropy 5.x+", but maybe not so much if that is a non-released version? 🤷

Reply:
I added this to the end of the plugin's `__init__` but it's still showing the contents of the plugin:
```python
        if not minversion("astropy", "5.3.dev"):
            self.disabled_msg = "Spectral Extraction in Cubeviz requires astropy 5.3.dev"
```

Reply:
Oh I have to update .vue too, don't I.

Reply:
Yep, that was it. Thanks!

Reply:
> Oh I have to update .vue too, don't I.

Yes, should just need `:disabled_msg="disabled_msg"` in `j-tray-plugin`

Reply:
Done 👌🏻

Reply:
Re: not minversion -- Maybe we should define it as `ASTROPY_LT_5_3` in the same style as how astropy define minversion for their dependencies. That way, it is easy to grep and remove when we no longer require that check.